src/labs/hooks/useUpdateLab.ts

```diff
@@ -0,0 +1,17 @@
+import { useMutation, queryCache } from 'react-query'
+
+import LabRepository from '../../shared/db/LabRepository'
+import Lab from '../../shared/model/Lab'
+
+function updateLab(labToUpdate: Lab): Promise<Lab> {
+  return LabRepository.saveOrUpdate(labToUpdate)
+}
+
+export default function useUpdateLab() {
+  return useMutation(updateLab, {
+    onSuccess: async () => {
+      await queryCache.invalidateQueries('labs')
+      await queryCache.invalidateQueries('lab')
```

Comment:
rather than invalidating the query, we can optimistically update the completed lab with the new content.

react-query.tanstack.com/docs/guides/updates-from-mutation-responses